mlir/lib/Dialect/Vector/Transforms/LowerVectorTransfer.cpp

```diff
@@ -639,14 +643,15 @@ struct TransferWriteToVectorStoreLowering
                    << write;
             });
 
-      rewriter.replaceOpWithNewOp<vector::MaskedStoreOp>(
-          write, write.getSource(), write.getIndices(), write.getMask(),
-          write.getVector());
+      rewriter.create<vector::MaskedStoreOp>(
+          write.getLoc(), write.getSource(), write.getIndices(),
+          write.getMask(), write.getVector());
+      return Value();
```

Comment:
It is confusing when it returns `Value()`. I think it is better to return a `FailureOr<Operation *>` in `matchAndRewriteMaskableOp` method. Because we eventually will just replace the op with the new operation.

And perhaps we don't need the check anymore.

https://github.com/llvm/llvm-project/blob/d4ff9615a1531f4a466b7d1fb0f175e3ae489289/mlir/include/mlir/Dialect/Vector/Utils/VectorUtils.h#L160-L167

Reply:
+1 to your comment @hanhanW , but this is a bit more nuanced 😅 

I implemented it this because there's quite a few places that simply return `Value`, e.g.:
https://github.com/llvm/llvm-project/blob/77db8b08c8b186c2625f8dfb26bb976561b43c4c/mlir/lib/Dialect/Vector/Transforms/LowerVectorContract.cpp#L432-L455

Lambdas like that get used a lot :) I recall trying to update that and other similar examples, but never got round to it. Sounds like a worthwhile TODO!